Description of GatewayClass is nil gives error in gwctl describe

<!--
Thank you for your interest in Gateway API! Please note that bug reports
here should only be used for bugs with the API itself, such as:

- Incomplete or inaccurate validation
- Problems with the API specification
- Flaws in conformance tests

Bugs that don't fit into one of the above categories should likely be filed
with the [implementation](https://gateway-api.sigs.k8s.io/implementations/)
of Gateway API that you're using.

Please use this template while reporting a bug and provide as much info as
possible. Not doing so may result in your bug not being addressed in a timely
manner. Thank you!
-->

**What happened**:
I deployed `Kong Ingress Controller` on a Kind cluster. When I run `gwctl describe gatewayclasses` I get the following error
```shell
$ go run cmd/main.go describe gatewayclasses
panic: runtime error: invalid memory address or nil pointer dereference
[signal SIGSEGV: segmentation violation code=0x1 addr=0x8 pc=0x13ade0e]

goroutine 1 [running]:
sigs.k8s.io/gateway-api/gwctl/pkg/printer.(*GatewayClassesPrinter).PrintDescribeView(0xc000d07980, 0xc000f0c030)
	/home/devaanshk840/Code/gateway-api/gwctl/pkg/printer/gatewayclasses.go:53 +0x10e
sigs.k8s.io/gateway-api/gwctl/pkg/cmd/describe.runDescribe({0xc00100e0d0?, 0x0?, 0x0?}, 0x0?, 0x0?)
	/home/devaanshk840/Code/gateway-api/gwctl/pkg/cmd/describe/describe.go:121 +0x8bc
sigs.k8s.io/gateway-api/gwctl/pkg/cmd/describe.NewDescribeCommand.func1(0xc000c8eb00?, {0xc00100e0d0?, 0x1?, 0x16781b1?})
	/home/devaanshk840/Code/gateway-api/gwctl/pkg/cmd/describe/describe.go:46 +0x2c
github.com/spf13/cobra.(*Command).execute(0xc0001b1500, {0xc00100e0a0, 0x1, 0x1})
	/home/devaanshk840/Code/gateway-api/gwctl/vendor/github.com/spf13/cobra/command.go:987 +0xaa3
github.com/spf13/cobra.(*Command).ExecuteC(0xc0001b0300)
	/home/devaanshk840/Code/gateway-api/gwctl/vendor/github.com/spf13/cobra/command.go:1115 +0x3ff
github.com/spf13/cobra.(*Command).Execute(...)
	/home/devaanshk840/Code/gateway-api/gwctl/vendor/github.com/spf13/cobra/command.go:1039
main.main()
	/home/devaanshk840/Code/gateway-api/gwctl/cmd/main.go:68 +0x2a5
exit status 2
```

**Anything else we need to know?**:
This error is occuring at the following line of code https://github.com/kubernetes-sigs/gateway-api/blob/main/gwctl/pkg/printer/gatewayclasses.go#L53 because `gatewayClassNode.GatewayClass.Spec.Description` is a pointer to string and `Kong` does not set a description for its gateway class. So essentially `*gatewayClassNode.GatewayClass.Spec.Description` is trying to dereference a nil pointer.

/area gwctl
/kind bug

--- a/gwctl/pkg/printer/gatewayclasses.go
+++ b/gwctl/pkg/printer/gatewayclasses.go
@@ -97,9 +97,11 @@ func (gcp *GatewayClassesPrinter) PrintDescribeView(resourceModel *resourcedisco
 			{
 				ControllerName: string(gatewayClassNode.GatewayClass.Spec.ControllerName),
 			},
-			{
+		}
+		if gatewayClassNode.GatewayClass.Spec.Description != nil {
+			views = append(views, gatewayClassDescribeView{
 				Description: gatewayClassNode.GatewayClass.Spec.Description,
-			},
+			})
 		}
 
 		if policyRefs := resourcediscovery.ConvertPoliciesMapToPolicyRefs(gatewayClassNode.Policies); len(policyRefs) != 0 {

--- a/gwctl/pkg/printer/gatewayclasses.go
+++ b/gwctl/pkg/printer/gatewayclasses.go
@@ -20,7 +20,6 @@ import (
 	"fmt"
 	"io"
 	"os"
-
 	"sort"
 	"strings"
 	"text/tabwriter"
@@ -95,11 +94,19 @@ func (gcp *GatewayClassesPrinter) PrintDescribeView(resourceModel *resourcedisco
 			{
 				Name: gatewayClassNode.GatewayClass.GetName(),
 			},
-			{
+		}
+
+		if gatewayClassNode.GatewayClass.Spec.Description != nil {
+			views = append(views, gatewayClassDescribeView{
 				ControllerName: string(gatewayClassNode.GatewayClass.Spec.ControllerName),
 				Description:    *gatewayClassNode.GatewayClass.Spec.Description,
-			},
+			})
+		} else {
+			views = append(views, gatewayClassDescribeView{
+				ControllerName: string(gatewayClassNode.GatewayClass.Spec.ControllerName),
+			})
 		}
+
 		if policyRefs := resourcediscovery.ConvertPoliciesMapToPolicyRefs(gatewayClassNode.Policies); len(policyRefs) != 0 {
 			views = append(views, gatewayClassDescribeView{
 				DirectlyAttachedPolicies: policyRefs,

--- a/gwctl/pkg/printer/gatewayclasses.go
+++ b/gwctl/pkg/printer/gatewayclasses.go
@@ -43,7 +43,7 @@ type gatewayClassDescribeView struct {
 	Name           string `json:",omitempty"`
 	ControllerName string `json:",omitempty"`
 	// GatewayClass description
-	Description              string                 `json:",omitempty"`
+	Description              *string                `json:",omitempty"`
 	DirectlyAttachedPolicies []policymanager.ObjRef `json:",omitempty"`
 }
 
@@ -94,17 +94,12 @@ func (gcp *GatewayClassesPrinter) PrintDescribeView(resourceModel *resourcedisco
 			{
 				Name: gatewayClassNode.GatewayClass.GetName(),
 			},
-		}
-
-		if gatewayClassNode.GatewayClass.Spec.Description != nil {
-			views = append(views, gatewayClassDescribeView{
+			{
 				ControllerName: string(gatewayClassNode.GatewayClass.Spec.ControllerName),
-				Description:    *gatewayClassNode.GatewayClass.Spec.Description,
-			})
-		} else {
-			views = append(views, gatewayClassDescribeView{
-				ControllerName: string(gatewayClassNode.GatewayClass.Spec.ControllerName),
-			})
+			},
+			{
+				Description: gatewayClassNode.GatewayClass.Spec.Description,
+			},
 		}
 
 		if policyRefs := resourcediscovery.ConvertPoliciesMapToPolicyRefs(gatewayClassNode.Policies); len(policyRefs) != 0 {

--- a/gwctl/pkg/printer/gatewayclasses_test.go
+++ b/gwctl/pkg/printer/gatewayclasses_test.go
@@ -128,68 +128,58 @@ foo-com-internal-gateway-class  foo.com/internal-gateway-class  Unknown   24m
 
 func TestGatewayClassesPrinter_PrintDescribeView(t *testing.T) {
 	fakeClock := testingclock.NewFakeClock(time.Now())
-	objects := []runtime.Object{
-		&gatewayv1.GatewayClass{
-			ObjectMeta: metav1.ObjectMeta{
-				Name: "foo-gatewayclass",
-			},
-			Spec: gatewayv1.GatewayClassSpec{
-				ControllerName: "example.net/gateway-controller",
-				Description:    common.PtrTo("random"),
-			},
-		},
-		&apiextensionsv1.CustomResourceDefinition{
-			ObjectMeta: metav1.ObjectMeta{
-				Name: "healthcheckpolicies.foo.com",
-				Labels: map[string]string{
-					gatewayv1alpha2.PolicyLabelKey: "true",
+
+	testcases := []struct {
+		name    string
+		objects []runtime.Object
+		want    string
+	}{
+		{
+			name: "GatewayClass with description and policy",
+			objects: []runtime.Object{
+				&gatewayv1.GatewayClass{
+					ObjectMeta: metav1.ObjectMeta{
+						Name: "foo-gatewayclass",
+					},
+					Spec: gatewayv1.GatewayClassSpec{
+						ControllerName: "example.net/gateway-controller",
+						Description:    common.PtrTo("random"),
+					},
 				},
-			},
-			Spec: apiextensionsv1.CustomResourceDefinitionSpec{
-				Group:    "foo.com",
-				Versions: []apiextensionsv1.CustomResourceDefinitionVersion{{Name: "v1"}},
-				Names: apiextensionsv1.CustomResourceDefinitionNames{
-					Plural: "healthcheckpolicies",
-					Kind:   "HealthCheckPolicy",
+				&apiextensionsv1.CustomResourceDefinition{
+					ObjectMeta: metav1.ObjectMeta{
+						Name: "healthcheckpolicies.foo.com",
+						Labels: map[string]string{
+							gatewayv1alpha2.PolicyLabelKey: "true",
+						},
+					},
+					Spec: apiextensionsv1.CustomResourceDefinitionSpec{
+						Group:    "foo.com",
+						Versions: []apiextensionsv1.CustomResourceDefinitionVersion{{Name: "v1"}},
+						Names: apiextensionsv1.CustomResourceDefinitionNames{
+							Plural: "healthcheckpolicies",
+							Kind:   "HealthCheckPolicy",
+						},
+					},
 				},
-			},
-		},
-		&unstructured.Unstructured{
-			Object: map[string]interface{}{
-				"apiVersion": "foo.com/v1",
-				"kind":       "HealthCheckPolicy",
-				"metadata": map[string]interface{}{
-					"name": "policy-name",
-				},
-				"spec": map[string]interface{}{
-					"targetRef": map[string]interface{}{
-						"group": "gateway.networking.k8s.io",
-						"kind":  "GatewayClass",
-						"name":  "foo-gatewayclass",
+				&unstructured.Unstructured{
+					Object: map[string]interface{}{
+						"apiVersion": "foo.com/v1",
+						"kind":       "HealthCheckPolicy",
+						"metadata": map[string]interface{}{
+							"name": "policy-name",
+						},
+						"spec": map[string]interface{}{
+							"targetRef": map[string]interface{}{
+								"group": "gateway.networking.k8s.io",
+								"kind":  "GatewayClass",
+								"name":  "foo-gatewayclass",
+							},
+						},
 					},
 				},
 			},
-		},
-	}
-
-	params := utils.MustParamsForTest(t, common.MustClientsForTest(t, objects...))
-	discoverer := resourcediscovery.Discoverer{
-		K8sClients:    params.K8sClients,
-		PolicyManager: params.PolicyManager,
-	}
-	resourceModel, err := discoverer.DiscoverResourcesForGatewayClass(resourcediscovery.Filter{})
-	if err != nil {
-		t.Fatalf("Failed to construct resourceModel: %v", resourceModel)
-	}
-
-	gcp := &GatewayClassesPrinter{
-		Out:   params.Out,
-		Clock: fakeClock,
-	}
-	gcp.PrintDescribeView(resourceModel)
-
-	got := params.Out.(*bytes.Buffer).String()
-	want := `
+			want: `
 Name: foo-gatewayclass
 ControllerName: example.net/gateway-controller
 Description: random
@@ -197,8 +187,50 @@ DirectlyAttachedPolicies:
 - Group: foo.com
   Kind: HealthCheckPolicy
   Name: policy-name
-`
-	if diff := cmp.Diff(common.YamlString(want), common.YamlString(got), common.YamlStringTransformer); diff != "" {
-		t.Errorf("Unexpected diff\ngot=\n%v\nwant=\n%v\ndiff (-want +got)=\n%v", got, want, diff)
+`,
+		},
+		{
+			name: "GatewayClass with no description",
+			objects: []runtime.Object{
+				&gatewayv1.GatewayClass{
+					ObjectMeta: metav1.ObjectMeta{
+						Name: "foo-gatewayclass",
+					},
+					Spec: gatewayv1.GatewayClassSpec{
+						ControllerName: "example.net/gateway-controller",
+					},
+				},
+			},
+			want: `
+Name: foo-gatewayclass
+ControllerName: example.net/gateway-controller
+`,
+		},
+	}
+
+	for _, tc := range testcases {
+		tc := tc
+		t.Run(tc.name, func(t *testing.T) {
+			params := utils.MustParamsForTest(t, common.MustClientsForTest(t, tc.objects...))
+			discoverer := resourcediscovery.Discoverer{
+				K8sClients:    params.K8sClients,
+				PolicyManager: params.PolicyManager,
+			}
+			resourceModel, err := discoverer.DiscoverResourcesForGatewayClass(resourcediscovery.Filter{})
+			if err != nil {
+				t.Fatalf("Failed to construct resourceModel: %v", resourceModel)
+			}
+
+			gcp := &GatewayClassesPrinter{
+				Out:   params.Out,
+				Clock: fakeClock,
+			}
+			gcp.PrintDescribeView(resourceModel)
+
+			got := params.Out.(*bytes.Buffer).String()
+			if diff := cmp.Diff(common.YamlString(tc.want), common.YamlString(got), common.YamlStringTransformer); diff != "" {
+				t.Errorf("Unexpected diff\ngot=\n%v\nwant=\n%v\ndiff (-want +got)=\n%v", got, tc.want, diff)
+			}
+		})
 	}
 }

